### ISSUE TITLE
v0.28 : Import menu has disappeared

### ISSUE THREAD

**Issue by TheLastProject** [TheLastProject/Catima#59]
<a href="https://github.com/llange"><img src="https://avatars0.githubusercontent.com/u/222432?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [llange](https://github.com/llange)**
_Tuesday Mar 17, 2020 at 21:39 GMT_
_Originally opened as https://github.com/brarcher/loyalty-card-locker/issues/364_

----

Since I've switched to 0.28 I cannot import any more.
The import/export menu display the import/export page ; however this page only has the "Export" section + button - no more "Import" section.
Note:
- French locale
- Linux localhost 3.10.108-ga0fd64c #1 SMP PREEMPT Fri Jun 15 09:31:54 UTC 2018 armv7l
- Android 7.1.2 (samsung/lineage_kccat6/kccat6:7.1.2/NJH47F/4c77d2651b:userdebug/release-keys)
- LineageOS 14.1-20180615-NIGHTLY-kccat6

<img width="608" alt="image" src="https://user-images.githubusercontent.com/222432/76904391-29966680-68a0-11ea-9541-26d7bb1fc04b.png">

**Comment by TheLastProject** [comment]
<a href="https://github.com/TheLastProject"><img src="https://avatars1.githubusercontent.com/u/1885159?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TheLastProject](https://github.com/TheLastProject)**
_Tuesday Mar 17, 2020 at 21:44 GMT_

----

What options did you have in 0.27? This honestly quite surprises me, I really didn't expect an Android 7 device that would lack a way to answer the GET_CONTENT intent

**Comment by TheLastProject** [comment]
<a href="https://github.com/llange"><img src="https://avatars0.githubusercontent.com/u/222432?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [llange](https://github.com/llange)**
_Tuesday Mar 17, 2020 at 22:00 GMT_

----

Hello @TheLastProject , on 0.27 it looks as expected:

<img width="610" alt="image" src="https://user-images.githubusercontent.com/222432/76905849-189b2480-68a3-11ea-8f92-19a9a6115d6d.png">

**Comment by TheLastProject** [comment]
<a href="https://github.com/TheLastProject"><img src="https://avatars1.githubusercontent.com/u/1885159?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TheLastProject](https://github.com/TheLastProject)**
_Tuesday Mar 17, 2020 at 22:03 GMT_

----

Ah, I see. You were already missing some import options but I indeed broke it completely for you.

As a workaround you could install any file manager, but I'll look into fixing this at some point so you won't have to.

**Comment by TheLastProject** [comment]
<a href="https://github.com/TheLastProject"><img src="https://avatars1.githubusercontent.com/u/1885159?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TheLastProject](https://github.com/TheLastProject)**
_Tuesday Mar 17, 2020 at 22:12 GMT_

----

Oh, another thing.

Could you tell me if the export function works on your device? Because that uses the internal file manager now too. I'm trying to understand on what layer things go wrong.

**Comment by TheLastProject** [comment]
<a href="https://github.com/llange"><img src="https://avatars0.githubusercontent.com/u/222432?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [llange](https://github.com/llange)**
_Tuesday Mar 17, 2020 at 22:21 GMT_

----

@TheLastProject thanks a lot for your comments.
I do have (multiple) file managers here : among which I have: 
* https://f-droid.org/en/packages/com.veniosg.dir/
* LineageOS default file manager (I believe it's https://github.com/LineageOS/android_packages_apps_DocumentsUI )
* OI FileManager 2.2.3

In 0.28 I'm able to export : It launches the default file manager where I'm able to navigate and modify the file name, and then the export seems OK:

<img width="602" alt="image" src="https://user-images.githubusercontent.com/222432/76907190-e3dc9c80-68a5-11ea-8aa2-6589341f9fec.png">

Even if the filename looks strange, it has been created at the proper place with proper content.

**Comment by TheLastProject** [comment]
<a href="https://github.com/llange"><img src="https://avatars0.githubusercontent.com/u/222432?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [llange](https://github.com/llange)**
_Tuesday Mar 17, 2020 at 22:26 GMT_

----

Wait ! I found it.

I'm using XPrivacyLua, and starting from 0.28 it interacts by filtering some part of the import/export screen.

By disabling the checkbox "See installed applications" ; everything is now back and I can see the other options:

<img width="604" alt="image" src="https://user-images.githubusercontent.com/222432/76907507-a7f60700-68a6-11ea-8fcc-1a8a0e1e2143.png">

I'm sorry for the report - perhaps we may add a small info about this new behaviour in the README or the release notes ?

**Comment by TheLastProject** [comment]
<a href="https://github.com/TheLastProject"><img src="https://avatars1.githubusercontent.com/u/1885159?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TheLastProject](https://github.com/TheLastProject)**
_Wednesday Mar 18, 2020 at 05:23 GMT_

----

I'm so glad you found it and thanks for sharing what it was, it's very interesting to learn how XPrivacyLua interacts with things!

There may be ways this can be improved, yeah. I don't think mentioning XPrivacyLua in the README is really worth it but the app should probably deal more gracefully with no file manager being found.

**Comment by TheLastProject** [comment]
<a href="https://github.com/llange"><img src="https://avatars0.githubusercontent.com/u/222432?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [llange](https://github.com/llange)**
_Wednesday Mar 18, 2020 at 10:07 GMT_

----

Thanks a lot for your comment !

I'm suggesting closing it as a solution was found - may be you want to open a distinct (related) issue regarding the lack of file manager ? Or if you want you can keep this one open instead.

**Comment by TheLastProject** [comment]
<a href="https://github.com/TheLastProject"><img src="https://avatars1.githubusercontent.com/u/1885159?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TheLastProject](https://github.com/TheLastProject)**
_Wednesday Mar 18, 2020 at 17:02 GMT_

----

I think keeping this open would be fine, as it has more context. Unless @brarcher disagrees.

**Comment by TheLastProject** [comment]
<a href="https://github.com/brarcher"><img src="https://avatars1.githubusercontent.com/u/5264535?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [brarcher](https://github.com/brarcher)**
_Thursday Mar 19, 2020 at 05:29 GMT_

----

Keeping this open is fine. XPrivacyLua uses Xposed, and that is a bit down the rabbit hole to mention in the README.  The [CHANGELOG](https://github.com/brarcher/loyalty-card-locker/blob/master/CHANGELOG.md) does mention a change to importing/exporting.

**Comment by IzzySoft** [comment]
Just as a note: I just stumbled into the very same issue. Yes, with XPrivacyLua. Took me a little to figure what I had to permit, but it was the very same option.

As for that "rabbit hole": I tend to agree; those who use it are also used to such issues (with new apps, it's even the first "culprit" coming to mind for me). Though it's not that apparent that import would rely on "other applications". I basically figured that by principle of exclusion – as nothing else seemed to make sense, this one was left and I thought it might be some "file manager" call (even though the system should handle it here; I was rather expecting a built-in file picker).

**TL;DR:** the case might not as rare as you think – but those affected should be able to figure :smile:

**Comment by TheLastProject** [comment]
I think I should just remove the file manager check. I don't think there's any Android installs out there anymore without any file manager and if it does crash then, well, we'll have statistics on how common an eventual file manager may be. Unless trying to open the file manager will just crash with XPrivacyLua, no clue... It would be good to have at very least testing help from an XPrivacyLua user here.

**Comment by IzzySoft** [comment]
Maybe just catch the exception and go on (optionally showing a toast)? And as for testing help: Thankfully I've installed the APK signed by you – so why not make an APK with additional debug stuff in and let me have it, with some short instructions what you want me to focus on (like, "fully block with XPrivacyLua, open import dialog, save logcat from the open action – then repeat without block")?

**Comment by TheLastProject** [comment]
I've attached a debug .zip. I don't really know XPrivacyLua well so uhm... test everything you feel is relevant and tell me if there are issues? If all is fine I'll consider this fixed with 6d4babf944c6f316c65c15ce41943456f140d656 and will include it in the next release.

[app-debug.zip](https://github.com/TheLastProject/Catima/files/5573928/app-debug.zip)

**Comment by IzzySoft** [comment]
Cool idea to use a `.debug` suffix, so I could check it in parallel. Guess you won't need any logcat: everything worked fine. In case you're curious, I'm attaching it anyway. It's the "full segment", nothing removed (so you might see some unrelated stuff), covering the following events:

* Starting Catima
* opening the import/export menu
* tapping "from file system" (file picker was showing up fine)
* tapping "from external app" (empty list opened – I have no Dropbox etc installed; if it should have shown *anything* please let me know what, so I can tell you if it should have been available or not)

All this was done with XPrivacyLua in "deny all" mode (the default for fresh installed apps). So yes, I'd say this issue is solved by 6d4babf and can be closed.

[catima.log.gz](https://github.com/TheLastProject/Catima/files/5574191/catima.log.gz)

**Comment by TheLastProject** [comment]
Okay, so fixed in https://github.com/TheLastProject/Catima/commit/6d4babf944c6f316c65c15ce41943456f140d656 it is
